File: apps/web/public/scripts/create-wallet.py
#!/usr/bin/env python3
"""
Bout Network — Create EVM Wallet
Usage:
  1. pip install eth-account   (one-time)
  2. python3 create-wallet.py
"""

import os
import sys
import stat

BOUT_DIR = os.path.join(os.path.expanduser("~"), ".bout")
WALLET_FILE = os.path.join(BOUT_DIR, "wallet.env")

# Check if wallet already exists
if os.path.exists(WALLET_FILE):
    with open(WALLET_FILE, "r") as f:
        content = f.read()
    print(f"[Bout] Wallet already exists at {WALLET_FILE}")
    for line in content.strip().split("\n"):
        if line.startswith("BOUT_WALLET_ADDR="):
            print(f"[Bout] Address: {line.split('=', 1)[1]}")
    sys.exit(0)

# Try to import eth_account
try:
    from eth_account import Account
except ImportError:
    print("[Bout] Error: eth-account not found. Install it first:")
    print("  pip install eth-account")
    sys.exit(1)

# Generate wallet
account = Account.create()
private_key = account.key.hex()
address = account.address

# Ensure private key has 0x prefix
if not private_key.startswith("0x"):
    private_key = "0x" + private_key

# Save to ~/.bout/wallet.env
os.makedirs(BOUT_DIR, exist_ok=True)
with open(WALLET_FILE, "w") as f:
    f.write(f"BOUT_WALLET_KEY={private_key}\n")
    f.write(f"BOUT_WALLET_ADDR={address}\n")
os.chmod(WALLET_FILE, stat.S_IRUSR | stat.S_IWUSR)

print("[Bout] Wallet created successfully!")
print(f"[Bout] Address: {address}")
print(f"[Bout] Saved to: {WALLET_FILE}")
print()
print("Next: Get testnet USDC from https://faucet.circle.com (Base Sepolia)")

File: apps/web/public/scripts/create-wallet.mjs
#!/usr/bin/env node
/**
 * Bout Network — Create EVM Wallet
 * Usage:
 *   1. npm install viem   (one-time)
 *   2. node create-wallet.mjs
 *
 * Or one-liner:
 *   mkdir -p /tmp/bout-setup && cd /tmp/bout-setup && npm init -y > /dev/null 2>&1 && npm install viem > /dev/null 2>&1 && node /path/to/create-wallet.mjs
 */

import { existsSync, mkdirSync, writeFileSync, readFileSync, chmodSync } from 'fs'
import { homedir } from 'os'
import { join } from 'path'

const BOUT_DIR = join(homedir(), '.bout')
const WALLET_FILE = join(BOUT_DIR, 'wallet.env')

// Check if wallet already exists
if (existsSync(WALLET_FILE)) {
  const content = readFileSync(WALLET_FILE, 'utf-8')
  console.log('[Bout] Wallet already exists at', WALLET_FILE)
  const addr = content.match(/BOUT_WALLET_ADDR=(.+)/)?.[1]
  if (addr) console.log('[Bout] Address:', addr)
  process.exit(0)
}

// Dynamic import viem
let generatePrivateKey, privateKeyToAccount
try {
  const mod = await import('viem/accounts')
  generatePrivateKey = mod.generatePrivateKey
  privateKeyToAccount = mod.privateKeyToAccount
} catch {
  console.error('[Bout] Error: viem not found. Install it first:')
  console.error('  npm install viem')
  console.error('')
  console.error('Or run this one-liner:')
  console.error('  mkdir -p /tmp/bout-setup && cd /tmp/bout-setup && npm init -y > /dev/null 2>&1 && npm install viem > /dev/null 2>&1 && node create-wallet.mjs')
  process.exit(1)
}

// Generate wallet
const privateKey = generatePrivateKey()
const account = privateKeyToAccount(privateKey)

// Save to ~/.bout/wallet.env
mkdirSync(BOUT_DIR, { recursive: true })
writeFileSync(WALLET_FILE, [
  `BOUT_WALLET_KEY=${privateKey}`,
  `BOUT_WALLET_ADDR=${account.address}`,
  '',
].join('\n'))
chmodSync(WALLET_FILE, 0o600)

console.log('[Bout] Wallet created successfully!')
console.log('[Bout] Address:', account.address)
console.log('[Bout] Saved to:', WALLET_FILE)
console.log('')
console.log('Next: Get testnet USDC from https://faucet.circle.com (Base Sepolia)')
